packages/config/src/ProfInfoErr.ts

```diff
@@ -0,0 +1,95 @@
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+// for imperative operations
+import { IProfInfoErr } from "./doc/IProfInfoErr";
+import { ImperativeError, IImperativeError } from "../../error";
+import { IImperativeErrorParms } from "../../error/src/doc/IImperativeErrorParms";
+
+/**
+ * This class is the error exception mechanism for the ProfileInfo API.
+ * It is derived from ImperativeError. We use a separate class so that
+ * our consumer can check the type of error, and then rely on errorCode
+ * values that are unique to ProfInfoErr. ProfInfoErr will always
+ * populate the errorCode property. Our consumer can use the errorCode to
+ * determine if it should process partial results.
+ */
+export class ProfInfoErr extends ImperativeError {
+
+    // _______________________________________________________________________
+    // The following are the complete set of errorCodes for ProfInfoErr.
+
+    /**
+     * Unable to retrieve the schema from a URL reference.
+     * Currently, the ProfiInfo API does not attempt to retrieve a schema
+     * through a URL. A URL does work to provide intellisense in VSCode
+     * when editing a config file.
+     */
+    public static readonly CANT_GET_SCHEMA_URL: string = "CantGetSchemaUrl";
+
+    /**
+     * The specified type of profile location is invalid for the requested operation.
+     */
+    public static readonly INVALID_PROF_LOC_TYPE: string = "InvalidProfLocType";
+
+    /**
+     * Failed to load the schema for a specified type of profile.
+     */
+    public static readonly LOAD_SCHEMA_FAILED: string = "LoadSchemaFailed";
+
+    /**
+     * A required profile property was not assigned a value.
+     */
+    public static readonly MISSING_REQ_PROP: string = "MissingProp";
+
+    /**
+     * The ProfileInfo.readProfilesFromDisk function was not called before
+     * a function which requires that prerequisite.
+     */
+    public static readonly MUST_READ_FROM_DISK: string = "MustReadFromDisk";
+
+    /**
+     * A specified property that is expected to exist in a specified profile
+     * does not exist in that profile.
+     */
+    public static readonly PROP_NOT_IN_PROFILE: string = "PropNotInProfile";
+
+    // _______________________________________________________________________
+
+    /**
+     * Construct the ProfInfoErr error object. It adds properties in
+     * IProfInfoErr to the existing properties of ImperativeError.
+     *
+     * @param impErrDetails
+     *        ImperativeError details and text (stack, messages, etc.)
+     *
+     * @param impErrParms
+     *        ImperativeError control parameters to indicate logging of node-report and more
+     */
+    constructor(
+        impErrDetails: IImperativeError,
+        profErrProps?: IProfInfoErr,
+        impErrParms?: IImperativeErrorParms
```

Comment:
Rather than passing two key-value objects into the constructor, would it make sense for `IProfInfoErr` to extend the interface `IImperativeErrorParms` and pass only a single object?

Reply:
With no conflicting properties, I think it may make things a bit easier for the programmer 😋 

Not sure what the preferred word is between `props` and `parms` though

Reply:
I thought about doing this. The reason I did not, was that you needed to specify the ImperativeError details in a second nested level of the structure. If you do not specify either of the second two arguments (which is probably most cases), you would supply the same structure to ProfInfoErr, as you would to ImperativeError.

That being said, since others feel that the combined structure is a better interface, I am all for it. I will make the changes and push an update to the PR.

Reply:
Perhaps I misinterpreted the original comment. Were you suggesting keeping what ImperativeError calls "details" as the first argument to ProInfoErr, and only combining the second two arguments to ProfiInfoErr. That is, combining stuff specific to ProfileInfo and what ImperativeError calls its parms argument?

Reply:
I was thinking to replace:
`export interface IProfInfoErr {`
with:
`export interface IProfInfoErr extends IImperativeErrorParms {`

As long as the properties added to `IProfInfoErr` do not conflict with the `IImperativeErrorParms` properties, there should be no additional nesting of properties.